scalableminds/webknossos
# Shared Chunk Cache for all DatasetArrays, CacheWeight for AlfuCache

- The cache for remote dataset array contents can now have a configured size in bytes. New config option `datastore.cache.imageArrayChunks.maxSizeBytes`. Default is 2 GB, consider increasing for production.
- For this, one cache object is shared between the dataset arrays
- Switches from akka cache to Caffeine directly (akka uses caffeine internally as well), allowing to add cache item weighing (weight is now set to bytes per chunk for array chunks)
- Unified some caching usages / naming (no more distinction between AlfuCache and AlfuFoxCache)

### TODO
 - [x] use shared cache also in Precomputed + Neuroglacner classes
 - [x] include vault path in cache key
 - [x] how to clear cache for individual datasets?
 - [x] null pointer exception for loading volume annotation buckets


### URL of deployed dev instance (used for testing):
- https://cacheweight.webknossos.xyz

### Steps to test:
- Load some datasets, should still show
- annotate some, should still show
- use reload button and look for logging (cache chunks should be removed, but only for the reloaded layer)

### Issues:
- fixes #6630 
- contributes to #6969 

------
